bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java

```diff
@@ -0,0 +1,419 @@
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.openhab.binding.volumio.internal.mapping.VolumioData;
+import org.openhab.binding.volumio.internal.mapping.VolumioEvents;
+import org.openhab.binding.volumio.internal.mapping.VolumioServiceTypes;
+import org.openhab.core.library.types.NextPreviousType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.PlayPauseType;
+import org.openhab.core.library.types.RewindFastforwardType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.socket.client.Socket;
+import io.socket.emitter.Emitter;
+
+/**
+ * The {@link VolumioHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Patrick Sernetz - Initial Contribution
+ * @author Chris Wohlbrecht - Adaption for openHAB 3
+ * @author Michael Loercher - Adaption for openHAB 3
+ */
+@NonNullByDefault
+public class VolumioHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(VolumioHandler.class);
+
+    private @Nullable VolumioService volumio;
+
+    private final VolumioData state = new VolumioData();
+
+    public VolumioHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("channelUID: {}", channelUID);
+
+        if (volumio == null) {
+            logger.debug("Ignoring command {} = {} because device is offline.", channelUID.getId(), command);
+            if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "device is offline");
+            }
+            return;
+        }
+
+        try {
+            switch (channelUID.getId()) {
+                case VolumioBindingConstants.CHANNEL_PLAYER:
+                    handlePlaybackCommands(command);
+                    break;
+                case VolumioBindingConstants.CHANNEL_VOLUME:
+                    handleVolumeCommand(command);
+                    break;
+
+                case VolumioBindingConstants.CHANNEL_ARTIST:
+                    break;
+                case VolumioBindingConstants.CHANNEL_ALBUM:
+                    break;
+                case VolumioBindingConstants.CHANNEL_TRACK_TYPE:
+                    break;
+                case VolumioBindingConstants.CHANNEL_TITLE:
+                    break;
```

Comment:
You can use a fall through here:

```suggestion
                case VolumioBindingConstants.CHANNEL_ARTIST:
                case VolumioBindingConstants.CHANNEL_ALBUM:
                case VolumioBindingConstants.CHANNEL_TRACK_TYPE:
                case VolumioBindingConstants.CHANNEL_TITLE:
                    break;
```

Reply:
solved